doc/source/train/huggingface-accelerate.rst

```diff
@@ -3,31 +3,201 @@
-just as you would if you were running Accelerate without Ray.
+    def train_func(config):
+        # Instantiate the accelerator
+        accelerator = Accelerator(...)
 
-.. note::
-    ``AccelerateTrainer`` will override some settings set with ``accelerate config``, mainly related to
-    the topology and networking. See the :class:`AccelerateTrainer <ray.train.huggingface.AccelerateTrainer>`
-    API reference for more details.
+        model = ...
+        optimizer = ...
+        train_dataloader = ...
+        eval_dataloader = ...
+        lr_scheduler = ...
 
-Aside from Accelerate support, the usage is identical to :class:`TorchTrainer <ray.train.torch.TorchTrainer>`, meaning you define your own training function
-and use the :func:`~ray.train.report` API to report metrics, save checkpoints etc.
+        # Prepare everything for distributed training
+        (
+            model,
+            optimizer,
+            train_dataloader,
+            eval_dataloader,
+            lr_scheduler,
+        ) = accelerator.prepare(
+            model, optimizer, train_dataloader, eval_dataloader, lr_scheduler
+        )
 
+        # Start training
+        ...
+    
+    from ray.train.torch import TorchTrainer
+    from ray.train import ScalingConfig
 
-.. dropdown:: Code example
+    trainer = TorchTrainer(
+        train_func,
+        scaling_config=ScalingConfig(...),
+        ...
+    )
+    trainer.fit()
 
-    .. literalinclude:: ./doc_code/accelerate_trainer.py
+.. tip::
+
+    Unlike native PyTorch, Lightning or Transformers, you **don't need to** provide any additional Ray Train utilities 
+    like :meth:`~ray.train.torch.prepare_model` or :meth:`~ray.train.torch.prepare_data_loader` in your training funciton. Instead, 
+    you can just instantiate an `Accelerator <https://huggingface.co/docs/accelerate/main/en/package_reference/accelerator#accelerate.Accelerator>`_ 
+    and call `Accelerator.prepare() <https://huggingface.co/docs/accelerate/main/en/package_reference/accelerator#accelerate.Accelerator.prepare>`_ 
+    as usual to prepare everything for distributed training.
+
+How to setup Accelerate config
+------------------------------
+
+Hugging Face Accelerate provides two approaches to configure your run:
+
+1. Use `accelerate config` to generate a configuration YAML file, and use `accelerate launch $YOUR_PY_SCRIPT --config_file=$YOUR_CONFIG_YAML` to launch via CLI. 
+2. Set configurations through the `accelerate.Accelerator <https://huggingface.co/docs/accelerate/main/en/package_reference/accelerator#accelerate.Accelerator>`_ object in your Python script, then run your script with a custom launcher.
+
+In Ray Train, you should always use the second approach, where Ray TorchTrainer acts as a distributed launcher and automatically sets up 
+the Torch distributed environment and run thr training function on all workers.
```

Comment:
Make sense. I can remove the enumerate points, and focus on discussing the second approach. And add an notice for accelerate config at the end of this section?

Reply:
Yeah, no need to make it stand out though. I think the point to call out is that you don't need the CLI command since TorchTrainer configures the launching part for you.